### PR TITLE
Launch dependencies before server unit tests

### PR DESCRIPTION
**Pull request by svix-daniel**
Launches PostgreSQL and Redis via the  `docker-compose.yml` for the server before running unit tests.

### CODE REVIEW

File: .github/workflows/server-ci.yml [hunk 1]
```diff
@@ -50,12 +52,24 @@ jobs:
         override: true
         profile: minimal
     - uses: Swatinem/rust-cache@v1
+
+    - name: Start dependencies
+      run: docker-compose -f "server/docker-compose.yml" up -d --build
+
```
Comment (svix-frank) [inline .github/workflows/server-ci.yml:58]: Does the github actions linux have docker & docker-compose by default? I don't see an install step

Comment (tasn) [reply .github/workflows/server-ci.yml:58]: I think you need to maybe say "uses docker", though given that the tests pass, I'm inclined to say the answer is yes? :)

File: server/testing-docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    healthcheck:
+      test: "pg_isready -h localhost"
+      interval: 30s
+      timeout: 10s
+      retries: 3
+    environment:
+      DB_HOST: "postgres"
+      DB_USER: "postgres"
+      DB_PASSWORD: "postgres"
+      MAX_CLIENT_CONN: 500
```
Comment (tasn) [inline server/testing-docker-compose.yml:21]:
```suggestion
```
No need for pgbouncer in the tests

File: .github/workflows/server-ci.yml (hunk 1 above)
Comment (svix-daniel) [reply .github/workflows/server-ci.yml]: `docker-compose` is installed in the actions image as shown here: https://github.com/actions/virtual-environments/blob/c8b63063441fb63048458f7907292a814c25189f/images/linux/ubuntu2004.json#L181

**Review by tasn — APPROVED** [review]

**Pull request merged**